lib/rex/socket/parameters.rb

```diff
@@ -383,7 +380,27 @@ def ssl
 
   # What version of SSL to use (Auto, SSL2, SSL3, SSL23, TLS1)
   # @return [String,Symbol]
-  attr_accessor :ssl_version
+  attr_reader :ssl_version
+  def ssl_version=(version)
+    # Let the caller specify a particular SSL/TLS version
+    case version
+    when 'SSL2'
+      version = :SSLv2
+    # 'TLS' will be the new name for autonegotation with newer versions of OpenSSL
+    when 'SSL23', 'TLS', 'Auto'
+      version = :SSLv23
+    when 'SSL3'
+      version = :SSLv3
+    when 'TLS1','TLS1.0'
+      version = :TLSv1
+    when 'TLS1.1'
+      version = :TLSv1_1
+    when 'TLS1.2'
+      version = :TLSv1_2
+    end
+
+    @ssl_version = version
```

Comment:
Do we want to set this to some default value if the value provided in `version` doesn't match anything? Or is this done cause the code in `lib/rex/socket/ssl.rb` will end up checking if the `version` value is supported or not?

Reply:
This is to maintain the behavior where the version is actually checked when it's being passed to the new OpenSSL context. It's checked here https://github.com/rapid7/rex-socket/pull/37/files#diff-61d855383234bf32c18b73d89c180383f72d4eb4cef37457a13e6eb7fc328327R130. The real acceptable values are the symbol ones like `:TLSv1_2` instead of the string value `'TLS1.2'`. The intention of this block is to keep that mapping in once place where it can be used by both client and server sockets. The parameters that Metasploit passes in here for example are the string values that need to be converted to OpenSSL's special symbols.

I think it'd be better to raise an exception here when the value isn't acceptable but that'd be a breaking change and not something I want to get into.

Reply:
Makes sense and that was the logic I thought was going on r.e the check so thanks for confirming this and explaining it more. Will mark this as resolved but I agree it would be nice to fix this but doing so would likely break key functionality without additional scoping and considerations.